apps/scratch/src/utilities/scratch-student-activities/pipeline.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  CreatePipelineParams,
+  DeletePipelineParams,
+  MovePipelineParams,
+  StudentAction,
+} from "../../types/scratch-student-activities";
+
+import {
+  trackCreateActivity,
+  trackMoveActivity,
+  trackDeleteActivity,
+} from "./index";
+
+type StudentActivityPipelineParams =
+  | CreatePipelineParams
+  | MovePipelineParams
+  | DeletePipelineParams;
+
+export const processStudentActivityPipeline = (
+  params: StudentActivityPipelineParams,
+): void => {
+  switch (params.action) {
+    case StudentAction.Create:
+      trackCreateActivity({
+        block: params.block,
+        sendRequest: params.sendRequest,
+        solution: params.solution,
+        event: params.event,
+      });
```

Comment:
I would suggest making all those functions directly accept `StudentActivityHandlerParams` and simply pass `params` here.